### ISSUE TITLE
Only shows when the key is held or set to Toggle Sprint (no double tap to sprint, no default mode)

### ISSUE THREAD

**Issue by CapoFantasma97** [theendercore/VisibleToggleSprint#1]
Ironically enough, in Minecraft the Hold Sprint mode is actually more akin of a regular Toggle Sprint in most games [MC-209971](https://bugs.mojang.com/browse/MC-209971). It probably mirrors the double tap Forward to sprint behavior.

Pressing once sets you into sprint mode, and you don't need to hold the key; you stop sprinting when hitting an obstacle or stop moving.
Toggle Sprint instead works like a sticky key, it will mantain the set mode (sprint/walk) even when stopping and resuming movement.

**This mod only shows the sprint indicator when the key is held, or if Toggle Sprint does it for you. It doesn't show the indicator when double tapping Forward to sprint or when sprinting without holding the key.**

**TL:DR** It would be nice if the mod could be adapted to show the indicator whenever the player is sprinting, regardless if using Toggle Sprint or not, because it seems to only register whether the game thinks the Sprint Key is held, even if you sprint without holding it or using it.

**Comment by theendercore** [comment]
I completely forgor about the fact you can double tap.
I'll look in to it.

**Comment by theendercore** [comment]
Also it's done this way cuz I have no idea to tell of the player is in sprint mode. mc code is just a gigantic mess at times and I don't know where to look for.

**Comment by CapoFantasma97** [comment]
Unfortunately I am not knowledgeable enough in the code part of the game.

In vanilla I'd use the scoreboard `stat.sprintOneCm` to detect sprinting. Not sure if this could help in finding the "root" of sprinting mechanic.

**Comment by theendercore** [comment]
Fixed in the latest update of the mod
